### PR TITLE
Some features (UPDATED 07.06)

### PR DESCRIPTION
**Pull request by blackMasterLord**
Add agents
Add ingame medal changer
Some fixes
Better Vote Revealer
**NEW!** Better Inventory Changer Menu
**NEW!** StatTrak™

### CODE REVIEW

**Comment by mkot2** [comment]
very nice

**Comment by wemanzoz** [comment]
very nic

**Comment by blackMasterLord** [comment]
> Can't add agents.

 fixed

File: Osiris/Hacks/InventoryChanger.cpp [hunk 1]
```diff
@@ -223,12 +226,20 @@ class StaticData {
...
                 if (const auto image = item->getInventoryImage()) {
                     _collectibles.emplace_back(isOriginal);
                     _gameItems.emplace_back(Type::Collectible, item->getRarity(), item->getWeaponId(), _collectibles.size() - 1, image);
                 }
-            } else if (itemTypeName == "#CSGO_Tool_Name_TagTag") {
+            }
+            else if (item->getCapabilities() & ITEM_CAP_CAN_PATCH) {
+                if (const auto image = item->getInventoryImage()) {
+                    _paintKits.emplace_back(0, L"");
+                    _gameItems.emplace_back(Type::Agent, item->getRarity(), item->getWeaponId(), _paintKits.size() - 1, image);
```
Comment (danielkrupinski) [inline Osiris/Hacks/InventoryChanger.cpp:239]: `_paintKits.emplace_back(0, L"");` is for adding "vanilla" paint kits. Agents do not use paint kits at all so remove that line and replace `_paintKits.size() - 1` with zero.

**Review by danielkrupinski — CHANGES_REQUESTED** [review]

**Comment by Xonn1com** [comment]
obraz
Error while building?

**Comment by blackMasterLord** [comment]
> obraz
> 
> Help needed

sorry, my bad

> obraz
> 
> Help needed

fixed

File: Osiris/Hooks.cpp [hunk 2]
```diff
@@ -305,6 +307,52 @@ static void __STDCALL frameStageNotify(LINUX_ARGS(void* thisptr,) FrameStage sta
     hooks->client.callOriginal<void, 37>(stage);
 }
 
+static void __STDCALL dispatchUserMessage(int messageType, uint32_t arg, uint32_t length, void* data) noexcept
```
Comment (mkot2) [inline Osiris/Hooks.cpp:310]: does this work on linux?

Comment (blackMasterLord) [reply Osiris/Hooks.cpp:310]: maybe.
idk

**Comment by Asviix** [comment]
I'm guessing Medal only affects the visual of Client ? still doing god's work

**Comment by blackMasterLord** [comment]
> I'm guessing Medal only affects the visual of Client ? still doing god's work

client side only

**Comment by wemanzoz** [comment]
csgo_pwkKXiwN06
knife logo on kill message doesnt change, whilst it works on normal osiris. pls fix @blackMasterLord

**Comment by blackMasterLord** [comment]
> csgo_pwkKXiwN06
> knife logo on kill message doesnt change, whilst it works on normal osiris. pls fix @blackMasterLord

fixed

**Comment by wemanzoz** [comment]
@blackMasterLord since dk added agent patches. remove or nah

**Comment by wirekuro** [comment]
Dude can you make an other pr with only the Better Vote Revealer?

**Comment by uzdiiilekk** [comment]
Can someone send me dll of this completed ? everytime i get error.. my discord is - uzdiiilekk_#3312

**Comment by wirekuro** [comment]
https://github.com/blackMasterLord/Osiris-Some-features here's the project, if u keep getting errors try updating visual studio.

**Comment by wirekuro** [comment]
> Can someone send me dll of this completed ? everytime i get error.. my discord is - uzdiiilekk_#3312

https://github.com/blackMasterLord/Osiris-Some-features here's the project, if u keep getting errors try updating visual studio.

**Comment by Alpha878** [comment]
@blackMasterLord do you think you are able to add lobby things and also troll options? Like you can see people locations, spam the lobby chat or send them 200 pop up errors?

**Comment by wemanzoz** [comment]
> @blackMasterLord is it possible for you to link me the branch for this pr?

can u instead make the better inv changer menu a seperate pr? @blackMasterLord

**Pull request closed without merging**